datalake/src/main/java/com/sequenceiq/datalake/flow/dr/restore/DatalakeRestoreActions.java

```diff
@@ -65,6 +69,46 @@
     @Inject
     private SdxService sdxService;
 
+    @Bean(name = "DATALAKE_TRIGGERING_RESTORE_STATE")
+    public Action<?, ?> triggerDatalakeRestore() {
+        return new AbstractSdxAction<>(DatalakeTriggerRestoreEvent.class) {
+            @Override
+            protected SdxContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> stateContext,
+                    DatalakeTriggerRestoreEvent payload) {
+                return SdxContext.from(flowParameters, payload);
+            }
+
+            @Override
+            protected void prepareExecution(DatalakeTriggerRestoreEvent payload, Map<Object, Object> variables) {
+                variables.put(OPERATION_ID, payload.getDrStatus().getOperationId());
+                super.prepareExecution(payload, variables);
+            }
+
+            @Override
+            protected void doExecute(SdxContext context, DatalakeTriggerRestoreEvent payload, Map<Object, Object> variables) {
+                LOGGER.info("Triggering datalake restore for {}", payload.getResourceId());
+                DatalakeDrStatusResponse restoreStatusResponse =
+                        sdxBackupRestoreService.triggerDatalakeRestore(payload.getResourceId(),
+                                payload.getBackupId(),
+                                payload.getBackupLocationOverride(),
+                                payload.getUserId());
+                variables.put(RESTORE_ID, restoreStatusResponse.getDrOperationId());
+                variables.put(OPERATION_ID, restoreStatusResponse.getDrOperationId());
+                payload.getDrStatus().setOperationId(restoreStatusResponse.getDrOperationId());
+                if (!restoreStatusResponse.failed()) {
+                    sendEvent(context, DatalakeDatabaseRestoreStartEvent.from(payload, restoreStatusResponse.getDrOperationId()));
+                } else {
+                    sendEvent(context, DATALAKE_RESTORE_FAILED_EVENT.event(), payload);
```

Comment:
pls log failure case

Reply:
Thanks. Added! Actually, when the failed event is handled by the handler, it will log the error and exception.